shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/route/engine/type/standard/ShardingStandardRoutingEngine.java

```diff
@@ -176,7 +176,7 @@ private boolean isGettingShardingValuesFromHint(final ShardingStrategy shardingS
     private List<ShardingConditionValue> getShardingValuesFromShardingConditions(final ShardingRule shardingRule, final Collection<String> shardingColumns, final ShardingCondition shardingCondition) {
         List<ShardingConditionValue> result = new ArrayList<>(shardingColumns.size());
         for (ShardingConditionValue each : shardingCondition.getValues()) {
-            Optional<BindingTableRule> bindingTableRule = shardingRule.findBindingTableRule(logicTableName);
+            Optional<BindingTableRule> bindingTableRule = shardingRule.findBindingTableRule(each.getTableName().toLowerCase());
```

Comment:
@setamv Why does it need to be lowercase?

Reply:
because the value of `each.getTableName()` is case sensitive，but table name in sharding rule is lower case.

Reply:
the table names are uppercase in out schema，and we config the binding rules with table name in uppercase like `- T_XX_MAIN_004,T_XX_DETAIL_004_EXT`, but the table names in `org.apache.shardingsphere.sharding.rule.ShardingRule#bindingTableRules` are converted to lowercase.
when we execute sql with table names in uppercase, the table name in ShardingConditionValue are kept in uppercase, so, i   must convert it's to lowercase the match the table name in sharding rule.

Reply:
> ShardingConditionValue

@setamv Thank you for reply.